pkg/controller/storagecluster/update.go

```diff
@@ -124,6 +144,38 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 	return c.syncNodes(cluster, oldPodsToDelete, []string{}, hash)
 }
 
+func getPxAPiPodForNode(nodeName string, ApiPods []*v1.Pod) *v1.Pod {
+
+	for _, pod := range ApiPods {
+		fmt.Println("PXPOD 2", pod.Spec.NodeName)
+		if reflect.DeepEqual(pod.Spec.NodeName, nodeName) {
+			return pod
+		}
+	}
+	return nil
+}
+
+func (c *Controller) getPxApiPods(cluster *corev1.StorageCluster) []*v1.Pod {
+	podList := &v1.PodList{}
+	err := c.client.List(context.TODO(),
+		podList,
+		&client.ListOptions{
+			Namespace:     cluster.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{"name": "portworx-api"}),
+		},
+	)
+
+	if err != nil {
+		return nil
+	}
+	result := make([]*v1.Pod, 0)
+	for _, pod := range podList.Items {
+		result = append(result, pod.DeepCopy())
+	}
+	return result
```

Comment:
Can we just return a map of `nodeName: podName`? We don't need the entire pod object anywhere.
It will avoid the unnecessary loops over the api pods above.